core/src/zxing/DecodeHints.cpp

```diff
@@ -29,7 +29,31 @@ using zxing::DecodeHints;
 // VC++
 using zxing::BarcodeFormat;
 
-const DecodeHintType DecodeHints::CHARACTER_SET;
```

Comment:
What happens if you just delete this line, and try compile? I mean without all the other changes you've made.